euphonic/qpoint_phonon_modes.py

```diff
@@ -1,57 +1,63 @@
+                      StructureFactor)
 
 
-class QpointPhononModes(object):
+T = TypeVar('T', bound='QpointPhononModes')
+
+
+class QpointPhononModes(QpointFrequencies):
     """
     A class to read and store vibrational data from model (e.g. CASTEP)
     output files
 
     Attributes
     ----------
-    crystal : Crystal
+    crystal
         Lattice and atom information
-    n_qpts : int
+    n_qpts
         Number of q-points in the object
-    qpts : (n_qpts, 3) float ndarray
-        Q-point coordinates, in fractional coordinates of the reciprocal
-        lattice
-    weights : (n_qpts,) float ndarray
-        The weight for each q-point
-    frequencies : (n_qpts, 3*crystal.n_atoms) float Quantity
-        Phonon frequencies per q-point and mode
-    eigenvectors : (n_qpts, 3*crystal.n_atoms, crystal.n_atoms, 3) complex ndarray
-        Dynamical matrix eigenvectors
+    qpts
+        Shape (n_qpts, 3) float ndarray. Q-point coordinates, in
+        fractional coordinates of the reciprocal lattice
+    weights
+        Shape (n_qpts,) float ndarray. The weight for each q-point
+    frequencies
+        Shape (n_qpts, 3*crystal.n_atoms) float Quantity. Phonon
+        frequencies per q-point and mode
+    eigenvectors
+        Shape (n_qpts, 3*crystal.n_atoms, crystal.n_atoms, 3) complex
+        ndarray. The dynamical matrix eigenvectors
     """
 
-    def __init__(self, crystal, qpts, frequencies, eigenvectors, weights=None):
+    def __init__(self, crystal: Crystal, qpts: np.ndarray,
+                 frequencies: Quantity, eigenvectors: Quantity,
+                 weights: Optional[np.ndarray] = None):
         """
         Parameters
         ----------
-        crystal : Crystal
+        crystal
             Lattice and atom information
-        qpts : (n_qpts, 3) float ndarray
-            Q-point coordinates
-        frequencies: (n_qpts, 3*crystal.n_atoms) float Quantity
-            Phonon frequencies, ordered according to increasing q-point
-            number. Default units meV
-        eigenvectors: (n_qpts, 3*crystal.n_atoms, crystal.n_atoms, 3) complex ndarray
-            Dynamical matrix eigenvectors
-        weights : (n_qpts,) float ndarray, optional
-            The weight for each q-point. If None, equal
-            weights are assumed
+        qpts
+            Shape (n_qpts, 3) float ndarray. The Q-point coordinates
+        frequencies
+            Shape (n_qpts, 3*crystal.n_atoms) float Quantity. Phonon
+            frequencies, ordered according to increasing q-point number.
+            Default units meV
+        eigenvectors
+            Shape (n_qpts, 3*crystal.n_atoms, crystal.n_atoms, 3)
+            complex ndarray. The dynamical matrix eigenvectors
+        weights
+            Shape (n_qpts,) float ndarray. The weight for each q-point.
+            If None, equal weights are assumed
         """
         _check_constructor_inputs(
```

Comment:
Could most of this `__init__` be replaced with `super().__init__(**args_except_for_eigenvectors)`?